.github/workflows/nuget.yml

```
@@ -42,29 +42,58 @@ jobs:
         useConfigFile: true
         additionalArguments: -updateprojectfiles
 
+    - name: Install AzureSignTool
+      run: dotnet tool install --global azuresigntool -v diag --ignore-failed-sources
+  
+    - name: Install NuGetKeyVaultSignTool
+      run: dotnet tool install --global NuGetKeyVaultSignTool -v diag --ignore-failed-sources
+
     - name: Build
       run: dotnet build Biosero.Orchestrator.DevTools.sln --configuration ${{ env.BUILD_CONFIGURATION }}
 
     - name: Test
       run: dotnet test Biosero.Orchestrator.DevTools.sln --no-build --configuration ${{ env.BUILD_CONFIGURATION }}
 
+    - name: Sign DLL
+      env:
+        AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+        AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+        AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+        CODE_SIGNING_CERT_NAME: ${{ vars.CODE_SIGNING_CERT_NAME }}
+        CODE_SIGNING_VAULT_URI: ${{ vars.CODE_SIGNING_VAULT_URI }}
+
+      run: |
+        # Install Azure CLI
+        curl -sL https://aka.ms/InstallAzureCLIDeb | powershell -Command -
+
+        # Authenticate with Azure using service principal
+        az login --service-principal -u ${{ secrets.AZURE_CLIENT_ID }} -p ${{ secrets.AZURE_CLIENT_SECRET }} --tenant ${{ secrets.AZURE_TENANT_ID }}
+
+        # Sign the DLL using Azure Sign Tool
+        AzureSignTool sign -c ${{ vars.CODE_SIGNING_CERT_NAME }} -v ${{ vars.CODE_SIGNING_VAULT_URI }} src/Biosero.Orchestrator.ScriptingTools/bin/${{ env.BUILD_CONFIGURATION }}/Biosero.Orchestrator.ScriptingTools.dll
```

Just run the single command like below. You do not need to login to azure to do this. Let the tool handle the credentials for you.

```suggestion
      AzureSignTool sign -kvu $env:CODE_SIGNING_VAULT_URI -kvi "$env:AZURE_CLIENT_ID" -kvt "$env:AZURE_TENANT_ID" -kvs "$env:AZURE_CLIENT_SECRET" -kvc $env:CODE_SIGNING_CERT_NAME -tr http://timestamp.digicert.com -v $file

```